dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/McpToolCallParams.java

```diff
@@ -71,32 +82,68 @@ public Map<String, Object> getArguments(Jsonb jsonb) {
         return parsedArguments;
     }
 
-    private Map<String, Object> parseArguments(JsonObject arguments, Jsonb jsonb) {
+    private Map<String, Object> parseArguments(JsonObject requestArguments, Jsonb jsonb) {
         Map<String, ArgumentMetadata> metadatas = metadata.arguments();
         Map<String, Object> result = new HashMap<>();
 
-        HashSet<String> argsProcessed = new HashSet<>();
-        for (var entry : arguments.entrySet()) {
-            String argName = entry.getKey();
-            JsonValue argValue = entry.getValue();
+        for (var argName : metadatas.keySet()) {
             ArgumentMetadata argMetadata = metadatas.get(argName);
```

Comment:
```suggestion
        for (var argEntry : metadatas.entrySet()) {
            String argName = argEntry.getKey();
            ArgumentMetadata argMetadata = argEntry.getValue();
```

Don't iterate on the keyset if you're going to look up the value for every key anyway.